src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -288,6 +288,10 @@ private function parseDefinition($id, $service, $file)
         }
 
         if (isset($service['decorates'])) {
+            if ('@' === $service['decorates']{0}) {
```

Comment:
and by the way, I don't think we use braces anywhere else to access chars in string (everywhere else we use square brackets instead IIRC)